### PR TITLE
feat(5b): add graphics finder with CPC pixel decode/encode/paint/export

### PR DESCRIPTION
**Pull request by ikari-pl**
## Summary
- Implements CPC pixel decoding for all three screen modes (0: 2px/byte, 1: 4px/byte, 2: 8px/byte)
- Provides encode/decode roundtrip (gfx_encode_byte reverses gfx_decode_byte exactly)
- Exports decoded graphics as 32-bit BMP files
- Paint individual pixels with gfx_paint (decode-modify-encode cycle)
- IPC commands: `gfx view`, `gfx decode`, `gfx paint`, `gfx palette`
- Reads live CPC palette from GateArray.ink_values + colours_rgb

## Test plan
- [ ] 25 unit tests covering all three CPC modes, roundtrip encoding, BMP export, paint, and edge cases
- [ ] Verify `gfx view 0xC000 40 25 1 /tmp/screen.bmp` exports a valid BMP
- [ ] Verify `gfx decode FF 0` returns correct pixel indices
- [ ] CI passes (macOS, Ubuntu, MINGW32, MINGW64)

### CODE REVIEW

File: src/gfx_finder.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    memcpy(info + 4, &width, 4);
+    int neg_height = -height;  // top-down BMP
+    memcpy(info + 8, &neg_height, 4);
+    uint16_t planes = 1;
+    memcpy(info + 12, &planes, 2);
+    uint16_t bpp = 32;
+    memcpy(info + 14, &bpp, 2);
+    // compression = 0 (BI_RGB), rest zeros
+    memcpy(info + 20, &data_size, 4);
+    fwrite(info, 1, 40, f);
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/gfx_finder.cpp:147]: Fixed in PR #31: BMP headers now use explicit little-endian byte writes (write_le32/write_le16 helpers) instead of memcpy of host-endian ints. Portable across big-endian systems.

File: src/gfx_finder.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        uint8_t p1 = ((val >> 6) & 1) | ((val >> 3) & 2) |
+                     ((val >> 0) & 4) | ((val << 3) & 8);
+        // WinAPE-compatible bit ordering: swap high/low nibble contribution
+        // The actual CPC hardware decodes: px0 bits {7,5,3,1}, px1 bits {6,4,2,0}
+        // Where bit7 is MSB of the color index and bit1 is LSB
+        out[0] = ((val & 0x80) >> 7) | ((val & 0x20) >> 4) |
+                 ((val & 0x08) >> 1) | ((val & 0x02) << 2);
+        out[1] = ((val & 0x40) >> 6) | ((val & 0x10) >> 3) |
+                 ((val & 0x04) >> 0) | ((val & 0x01) << 3);
+        (void)p0; (void)p1; // suppress unused variable warnings from initial attempt
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/gfx_finder.cpp:29]: Fixed in PR #31: Removed the dead p0/p1 variables and their (void) suppression.

File: src/gfx_finder.cpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            uint8_t r = (rgba >> 0) & 0xFF;
+            uint8_t g = (rgba >> 8) & 0xFF;
+            uint8_t b = (rgba >> 16) & 0xFF;
+            uint8_t a = (rgba >> 24) & 0xFF;
+            row[x * 4 + 0] = b;
+            row[x * 4 + 1] = g;
+            row[x * 4 + 2] = r;
+            row[x * 4 + 3] = a;
+        }
+        fwrite(row.data(), 1, static_cast<size_t>(row_bytes), f);
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/gfx_finder.cpp:163]: Fixed in PR #31: All fwrite calls now check return values and return false on write failure.
